data/modules/scripts/gamestore/init.lua

```diff
@@ -1,2268 +0,0 @@
-			GameStore.processMountPurchase(player, offer.id)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_NAMECHANGE then
-			local newName = msg:getString()
-			GameStore.processNameChangePurchase(player, offer, productType, newName)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_SEXCHANGE then
-			GameStore.processSexChangePurchase(player)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_EXPBOOST then
-			GameStore.processExpBoostPurchase(player)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HUNTINGSLOT then
-			GameStore.processTaskHuntingThirdSlot(player)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_PREYSLOT then
-			GameStore.processPreyThirdSlot(player)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_PREYBONUS then
-			GameStore.processPreyBonusReroll(player, offer.count)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_TEMPLE then
-			GameStore.processTempleTeleportPurchase(player)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_CHARGES then
-			GameStore.processChargesPurchase(player, offer.itemtype, offer.name, offer.charges, offer.movable, offer.setOwner)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING then
-			local hirelingName = msg:getString()
-			local sex = msg:getByte()
-			GameStore.processHirelingPurchase(player, offer, productType, hirelingName, sex)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING_NAMECHANGE then
-			local hirelingName = msg:getString()
-			GameStore.processHirelingChangeNamePurchase(player, offer, productType, hirelingName)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING_SEXCHANGE then
-			GameStore.processHirelingChangeSexPurchase(player, offer)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING_SKILL then
-			GameStore.processHirelingSkillPurchase(player, offer)
-		elseif offer.type == GameStore.OfferTypes.OFFER_TYPE_HIRELING_OUTFIT then
-			GameStore.processHirelingOutfitPurchase(player, offer)
-		else
-			-- This should never happen by our convention, but just in case the guarding condition is messed up...
-			error({ code = 0, message = "This offer is unavailable [2]" })
-		end
-	end)
-
-	if not pcallOk then
-		local alertMessage = pcallError.code and pcallError.message or "Something went wrong. Your purchase has been cancelled."
-
-		-- unhandled error
-		if not pcallError.code then
-			logger.warn("[parseBuyStoreOffer] - Purchase failed due to an unhandled script error. Stacktrace: {}", pcallError)
-		end
-
-		return queueSendStoreAlertToUser(alertMessage, 500, playerId)
-	end
-
-	if table.contains({ GameStore.OfferTypes.OFFER_TYPE_HOUSE, GameStore.OfferTypes.OFFER_TYPE_EXPBOOST, GameStore.OfferTypes.OFFER_TYPE_PREYBONUS, GameStore.OfferTypes.OFFER_TYPE_BLESSINGS, GameStore.OfferTypes.OFFER_TYPE_ALLBLESSINGS, GameStore.OfferTypes.OFFER_TYPE_INSTANT_REWARD_ACCESS }, offer.type) then
-		insertPlayerTransactionSummary(player, offer)
```

Comment:
this too.

Reply:
We don’t need all these additional calls; simply modifying the function in C++ will solve the issue. It’s quite straightforward.

Reply:
> We don’t need all these additional calls, simply modifying the function in C++ and call it will solve the issue. It’s quite straightforward.

when I said to create these functions and their calls. I mean creating it on cpp code.